src/DependenciesPackager/CrossgenContext.cs

```diff
@@ -110,78 +182,179 @@ public void FetchJitTools(string runtime, string restoreFolder)
                 throw new InvalidOperationException($"Cannot find executable {jitExecutable}.");
             }
 
-            var jitDir = Path.GetDirectoryName(jitExecutablePath);
-            foreach (var file in Directory.GetFiles(jitDir))
-            {
-                File.Copy(file, Path.Combine(_stagingFolder, Path.GetFileName(file)), overwrite: true);
-            }
+            _clrJitPath = jitExecutablePath;
         }
 
         public void Crossgen(string outputPath)
         {
-            foreach (var package in _packageEntries)
+            var lck = new object();
+            Parallel.ForEach(_packagesToCrossGen, package =>
             {
+                var packageDirectory = CreatePackageSubDirectory(outputPath, package);
+                CopyPackageSignature(package, outputPath);
+
                 foreach (var asset in package.Assets)
                 {
-                    var subDir = CreateSubDirectory(outputPath, package, asset);
+                    var subDir = CreateAssetSubDirectory(outputPath, package, asset);
                     var targetPath = Path.Combine(subDir.FullName, asset.FileName);
 
-                    if (!CrossgenAssembly(_stagingFolder, package, asset, targetPath))
+                    var crossGen = CrossgenAssembly(package, asset, targetPath);
+                    lock (lck)
```

Comment:
Just to print them in order, if not they can get intermingled and it makes the output hard to read (was happening on the console when I made it parallel))